src/ode_solver/rrk_explicit_ode_solver.cpp

```diff
@@ -0,0 +1,109 @@
+        } //sum(a_ij *k_j)
+        
+        this->rk_stage[i]*=dt; 
+        
+        this->rk_stage[i].add(1.0,this->solution_update); //u_n + dt * sum(a_ij * k_j)
+
+        this->dg->solution = this->rk_stage[i];
+        this->dg->assemble_residual(); //RHS : du/dt = RHS = F(u_n + dt* sum(a_ij*k_j))
+        this->dg->global_inverse_mass_matrix.vmult(this->rk_stage[i], this->dg->right_hand_side); //rk_stage[i] = IMM*RHS = F(u_n + dt*sum(a_ij*k_j))
+    }
+
+    real relaxation_parameter = compute_relaxation_parameter_explicit();
+    //assemble solution from stages
+    for (int i = 0; i < this->rk_order; ++i){
+       this->solution_update.add(
+               dt * relaxation_parameter * this->butcher_tableau_b[i],
+               this->rk_stage[i]); 
+    }
+    this->dg->solution = this->solution_update; // u_np1 = u_n +gamma*dt* sum(k_i * b_i)
+
+    ++(this->current_iteration);
+    this->current_time += relaxation_parameter * dt;
+}
+
+template <int dim, typename real, typename MeshType>
+real RRKExplicitODESolver<dim,real,MeshType>::compute_relaxation_parameter_explicit()
+{
+    double gamma = 1;
+    double denominator = 0;
+    double numerator = 0;
+    for (int i = 0; i < this->rk_order; ++i){
+        for (int j = 0; j < this->rk_order; ++j){
+            real inner_product = compute_inner_product(this->rk_stage[i],this->rk_stage[j]);
+            numerator += this->butcher_tableau_b[i] *this-> butcher_tableau_a[i][j] * inner_product; 
+            denominator += this->butcher_tableau_b[i]*this->butcher_tableau_b[j] * inner_product;
+        }
+    }
+    numerator *= 2;
+    gamma = (denominator < 1E-8) ? 1 : numerator/denominator;
+    return gamma;
+}
+
+template <int dim, typename real, typename MeshType>
+real RRKExplicitODESolver<dim,real,MeshType>::compute_inner_product (
+        dealii::LinearAlgebra::distributed::Vector<double> stage_i,
+        dealii::LinearAlgebra::distributed::Vector<double> stage_j
+        )
+{
+    // Intention is to point to physics (mimic structure in flow_solver_cases/periodic_turbulence.cpp for converting to solution for general nodes) 
+    // For now, only energy on collocated nodes is implemented.
+    
+    real inner_product = 0;
+
+    bool use_collocated_nodes = ODESolverBase<dim,real,MeshType>::all_parameters->use_collocated_nodes;
+    using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
+    PDEEnum pde_type = ODESolverBase<dim,real,MeshType>::all_parameters->pde_type;
+    bool use_inviscid_burgers = (pde_type == PDEEnum::burgers_inviscid);
+    
+    if (use_collocated_nodes && use_inviscid_burgers){
+        for (unsigned int i = 0; i < this->dg->solution.size(); ++i) {
+            inner_product += 1./(this->dg->global_inverse_mass_matrix.diag_element(i))
+                             * stage_i[i] * stage_j[i];
+                }
+    }else {
+        this->pcout << "RRK is only implemented for inviscid burgers and collocated nodes." << std::endl;
```

Comment:
Might be more appropriate for this to done at the factory level; at least to signal that RRK can't be used for `dim!=1`; I'd also comment out the appropriate/unused instantiations

Reply:
You're right, that's better to do in `ode_solver_factory`. Changing in the next commit.

Reply:
It looks like the unused instantiations are still there and there is still no safeguard for checking that `dim == 1`

Reply:
Yep, added a check for `dim=1` and commented out the extra instantiations.